src/nbl/video/ILogicalDevice.cpp

```diff
@@ -29,6 +29,62 @@ core::smart_refctd_ptr<IGPUDescriptorSetLayout> ILogicalDevice::createDescriptor
     return createDescriptorSetLayout_impl(_begin,_end);
 }
 
+bool ILogicalDevice::updateDescriptorSets(uint32_t descriptorWriteCount, const IGPUDescriptorSet::SWriteDescriptorSet* pDescriptorWrites, uint32_t descriptorCopyCount, const IGPUDescriptorSet::SCopyDescriptorSet* pDescriptorCopies)
+{
+    for (auto i = 0; i < descriptorWriteCount; ++i)
+    {
+        auto* ds = static_cast<IGPUDescriptorSet*>(pDescriptorWrites[i].dstSet);
+        ds->incrementVersion();
+
+        auto* descriptors = ds->getDescriptors(pDescriptorWrites[i].descriptorType, pDescriptorWrites[i].binding);
+        auto* samplers = ds->getMutableSamplers(pDescriptorWrites[i].binding);
+        for (auto j = 0; j < pDescriptorWrites[i].count; ++j)
+        {
+            descriptors[j] = pDescriptorWrites[i].info[j].desc;
+
+            if (samplers)
+                samplers[j] = pDescriptorWrites[i].info[j].info.image.sampler;
+        }
+    }
```

Comment:
since we need to declare `ILogicalDevice` as a friend of `IGPUDescriptorSet` we can wrap this all up into a `processWrites` method or something

then a `getDescriptors` and `getMutableSamplers` method probably doesn't need to exist

Reply:
If we get rid of `getDescriptors` and `getMutableSamplers` entirely and inline their contents instead, then there will be duplication, right now in `updateDescriptorSets`, or in the future in `processWrite`, and `processCopy`.